.buildkite/x-pack/pipeline.xpack.winlogbeat.yml

```diff
@@ -8,41 +8,156 @@ env:
+          - "x-pack/winlogbeat/build/*.xml"
+          - "x-pack/winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "Xpack/Winlogbeat Win-10 Unit Tests"
+
+      - label: ":windows: Xpack/Winlogbeat Win-11 Unit Tests"
+        command: |
+          Set-Location -Path x-pack/winlogbeat
+          mage build unitTest
+        key: "extended-win-11-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_11}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/winlogbeat/build/*.xml"
+          - "x-pack/winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "Xpack/Winlogbeat Win-11 Unit Tests"
+
+      - label: ":windows: Xpack/Winlogbeat Win-2019 Unit Tests"
+        command: |
+          Set-Location -Path x-pack/winlogbeat
+          mage build unitTest
+        key: "extended-win-2019-unit-tests"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_WIN_2019}"
+          machineType: "${GCP_WIN_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        artifact_paths:
+          - "x-pack/winlogbeat/build/*.xml"
+          - "x-pack/winlogbeat/build/*.json"
+        notify:
+          - github_commit_status:
+              context: "x-pack/winlogbeat: Win-2019 Unit Tests"
 
   - wait: ~
-    if: "build.source == 'ui'"
-    allow_dependency_failure: false
-
-  - label: ":linux: Load dynamic x-pack winlogbeat pipeline"
-    key: "xpack-winlogbeat-pipeline"
-    command: ".buildkite/scripts/generate_xpack_winlogbeat_pipeline.sh"
-    agents:
-      image: "docker.elastic.co/ci-agent-images/platform-ingest/buildkite-agent-beats-ci:latest"
-    notify:
-      - github_commit_status:
-          context: "${BEATS_PROJECT_NAME}: Load dynamic pipeline's steps"
+    # with PRs, we want to run packaging only if mandatory tests succeed
+    # for other cases, e.g. merge commits, we want to run packaging (and publish) independently of other tests
+    # this allows building DRA artifacts even if there is flakiness in mandatory tests
+    if: build.env("BUILDKITE_PULL_REQUEST") != "false"
+    depends_on:
+      - "xpack-winlogbeat-mandatory-tests"
+
+  - group: "Xpack/Winlogbeat Packaging"
+    key: "xpack-winlogbeat-packaging"
+
+    steps:
+      - label: ":ubuntu: Packaging Linux"
+        key: "packaging-linux"
+        command: "cd x-pack/winlogbeat && mage package"
+        agents:
+          provider: "gcp"
+          image: "${IMAGE_UBUNTU_X86_64}"
+          machineType: "${GCP_HI_PERF_MACHINE_TYPE}"
+          disk_size: 100
+          disk_type: "pd-ssd"
+        env:
+          PLATFORMS: "+all linux/amd64 linux/arm64 windows/amd64 darwin/amd64 darwin/arm64"
+        notify:
+          - github_commit_status:
+              context: "Xpack/Winlogbeat Packaging"
+
```

Comment:
As mentioned in a previous comment, I think larger refactorings/fixes for x-pack/winlogbeat (or any other specific beat) belong in a separate PR.

This PR should only be fixing common *small* issues that apply to all migrated beats so far.

Reply:
It was there since the branch was "growing" from x-pack/winlogbeat which wasn't in main at that moment. Rebase solved the issue :)